java/code/src/com/redhat/rhn/manager/errata/ErrataManager.java

```diff
@@ -252,6 +250,81 @@ public static Errata addChannelsToErrata(Errata errata,
+     * @param toChannel the target {@link Channel}
+     * @param fromChannel the source {@link Channel}
+     * @return the set of merged {@link Errata}
+     */
+    public static Set<Errata> mergeErrataToChannel(User user, Set<Errata> errataToMergeIn,
+            Channel toChannel, Channel fromChannel) {
+        return mergeErrataToChannel(user, errataToMergeIn, toChannel, fromChannel, false, true);
+    }
+
+    /**
+     * Merge given {@link Errata} from source {@link Channel} to target {@link Channel}.
+     *
+     * @param user User performing the operation
+     * @param errataToMergeIn set of {@link Errata} to merge
+     * @param toChannel the target {@link Channel}
+     * @param fromChannel the source {@link Channel}
+     * @param async run the merge asynchronously?
+     * @param repoRegen request regenerating repodata after merge?
+     * @return the set of merged {@link Errata}
+     */
+    public static Set<Errata> mergeErrataToChannel(User user, Set<Errata> errataToMergeIn,
+            Channel toChannel, Channel fromChannel, boolean async, boolean repoRegen) {
+        Set<Errata> errataToMerge = new HashSet<>(errataToMergeIn);
+
+        // find errata that we do not need to merge
+        List<Errata> same = listSamePublishedInChannels(user, fromChannel, toChannel);
+        List<Errata> brothers = listPublishedBrothersInChannels(user, fromChannel, toChannel);
+        List<Errata> clones = listPublishedClonesInChannels(user, fromChannel, toChannel);
+        // and remove them
+        errataToMerge.removeAll(same);
+        errataToMerge.removeAll(brothers);
+        errataToMerge.removeAll(clones);
+
+        log.debug("Publishing");
+        CloneErrataEvent eve = new CloneErrataEvent(toChannel, getErrataIds(errataToMerge), repoRegen, user);
+        if (async) {
+            MessageQueue.publish(eve);
+        }
+        else {
+            new CloneErrataAction().execute(eve);
+        }
+
+        // no need to regenerate errata cache, because we didn't touch any packages
+        return errataToMerge;
+    }
+
+    private static Set<Long> getErrataIds(Set<Errata> errata) {
+        Set<Long> ids = new HashSet<Long>();
+        for (Errata erratum : errata) {
+            ids.add(erratum.getId());
+        }
+        return ids;
+    }
+
+    /**
+     * Removes errata from target channel that are not present in the source channel.
+     *
+     * @param srcChannel the source channel
+     * @param tgtChannel the target channel
+     * @param user the user
+     */
+    public static void truncateErrata(Channel srcChannel, Channel tgtChannel, User user) {
+        // 2-pass approach to prevent concurrent modification error
+        Set<Errata> toRemove = tgtChannel.getErratas().stream()
+                .filter(e -> !srcChannel.getErratas().contains(e))
+                .collect(toSet());
+        toRemove.forEach(e -> removeErratumFromChannel(e, tgtChannel, user));
+        // todo cleanup orphans?
```

Comment:
Something to do instead of a `// todo` here?

Reply:
Created a downstream issue https://github.com/SUSE/spacewalk/issues/7427.